packages/flutter_dynamic_forms_components/lib/src/components/date_range_picker/date_range_widget.dart

```diff
@@ -0,0 +1,452 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class CustomCalendarView extends StatefulWidget {
+  final DateTime minimumDate;
+  final DateTime maximumDate;
+  final DateTime initialStartDate;
+  final DateTime initialEndDate;
+  final Function(DateTime, DateTime) startEndDateChange;
+
+  const CustomCalendarView(
+      {Key key,
+      this.initialStartDate,
+      this.initialEndDate,
+      this.startEndDateChange,
+      this.minimumDate,
+      this.maximumDate})
+      : super(key: key);
+
+  @override
+  _CustomCalendarViewState createState() => _CustomCalendarViewState();
+}
+
+class _CustomCalendarViewState extends State<CustomCalendarView> {
+  List<DateTime> dateList = List<DateTime>();
+  var currentMonthDate = DateTime.now();
+  DateTime startDate;
+  DateTime endDate;
+
+  @override
+  void initState() {
+    setListOfDate(currentMonthDate);
+    if (widget.initialStartDate != null) {
+      startDate = widget.initialStartDate;
+    }
+    if (widget.initialEndDate != null) {
+      endDate = widget.initialEndDate;
+    }
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  void setListOfDate(DateTime monthDate) {
+    dateList.clear();
+    var newDate = DateTime(monthDate.year, monthDate.month, 0);
```

Comment:
I found the issue.
This will fix it:
`var newDate = DateTime(monthDate.year, monthDate.month, 0, 12);`
The problem is that I am testing it in the Czech Republic and I suppose you are located in Turkey. As I found out Turkey has switched to permanent summertime, whereas in European countries we set our clocks one hour back on Sunday 28. October. When you create the date without the number 12 you are starting the initial date at midnight. This means when the hour shifts back on sunday it will jump back to the previous day at 23 PM so that's why I am seeing shifted dates.

But I am not sure if you don't need to clear the hour value again when you are setting the date to the component model, because you probably want to set it to 0 AM again.

Reply:
Ohh yes sure why you see this problem. Maybe this will fix about time issue

var newDate = DateTime(monthDate.year, monthDate.month, 0).toUtc();

Reply:
Great idea with the UTC!
I think this is even better because it starts at midnight and doesn't shift hours:
`var newDate = DateTime.utc(monthDate.year, monthDate.month, 0);`

Reply:
Ok, chage it with DateTime.utc! Now everythings looks fine.